src/pure-assert.js

```diff
@@ -55,40 +55,118 @@ export class UserError extends Error {
  * Throws a provided error if the second argument isn't trueish.
  * @param {Object} errorCls
  * @param {T} shouldBeTruthy
- * @param {string} message
+ * @param {string} opt_message
```

Comment:
I'd just name the file `src/assert.js`. Another consideration here: I'd also consider the following separation that might work better for us:

1. `dev.js` - all dev asserts and other utils. This whole module is completely stripped from the PROD - nothing remains at all. Easy to check that this module is not part of any compilation.
2. `user.js` - all user-related utils to assert/log.

I guess what I'm really wondering here: why try to keep `devAssert` and `userAssert` together. Besides the fact that they both throw exceptions, is there anything else that they share? Another benefit: if we focus on `devAssert` we might be able to get through all of them quicker then `userAssert` and co.

/cc @jridgewell

Reply:
re: renaming, I was planning to do that in a follow-up, but it probably makes sense to do sooner rather than later. Will add here

Reply:
> I'd just name the file `src/assert.js`
Done

re: alternate split, it's not clear to me how much dev.js/user.js would own. I'd almost rather see something like a closure compiler comment/directive like `@devOnly` that lets helpers live in the appropriate module with related logic. Curious to hear Justin's thoughts

Reply:
The critical part is `devAssert` is basically covering for cases that compiler is not sufficient. When compiler has done its job, the `devAsserts` should disappear as they do now. I think this is a lot better than `@devOnly` since it's still a real code, just not something we want in the prod binary.